verde/utils.py

```diff
@@ -201,3 +202,93 @@ def maxabs(*args):
     arrays = [np.atleast_1d(i) for i in args]
     absolute = [np.abs([i.min(), i.max()]).max() for i in arrays]
     return np.max(absolute)
+
+
+def grid_to_table(grid):
+    """
+    Convert a grid to a table with the values and coordinates of each point.
+
+    Takes a 2D grid as input, extracts the coordinates and runs them through
+    :func:`numpy.meshgrid` to create a 2D table. Works for 2D grids and
+    n-dimensional variables. Use cases includes passing gridded data to
+    functions that expect data in XYZ format such as :class:`verde.BlockReduce`
+
+    Parameters
+    ----------
+    grid : :class:`xarray.Dataset`
+        A 2D grid with a single data variable.
+
+    Returns
+    -------
+    table : :class:`pandas.DataFrame`
+        :class:`pandas.DataFrame` with coordinates and variables.
+
+    Examples
+    --------
```

Comment:
In general, it's best to use realist sounding variable names, like "easting" instead of "dim_0" or "temperature" instead of "var1". That helps keep things clear to the user, who will probably have variables like that in their code.